### PR TITLE
[Ldap] [5.0] add type-hint to interface and implementation

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | 5.0
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | Contribute to #32179 
| License       | MIT
| Doc PR        |  

This adds the type-hint to the Ldap interfaces and implementation

### CODE REVIEW

File: src/Symfony/Component/Ldap/Adapter/AdapterInterface.php [hunk 1]
```diff
@@ -45,10 +45,10 @@ public function getEntryManager();
      * Escape a string for use in an LDAP filter or DN.
      *
      * @param string $subject
-     * @param string $ignore
+     * @param null|string $ignore
```
Comment (Simperfit) [inline src/Symfony/Component/Ldap/Adapter/AdapterInterface.php]: we pass null values in the test, I think it's safe to say that is nullable.

File: src/Symfony/Component/Ldap/Adapter/AdapterInterface.php [hunk 2]
```diff
@@ -44,11 +44,11 @@ public function getEntryManager();
     /**
      * Escape a string for use in an LDAP filter or DN.
      *
-     * @param string $subject
-     * @param string $ignore
-     * @param int    $flags
+     * @param string      $subject
+     * @param string|null $ignore
+     * @param int         $flags
```
Comment (derrabus) [inline src/Symfony/Component/Ldap/Adapter/AdapterInterface.php]: The `@param` annotations are redundant now and may be removed.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Ldap/Adapter/AdapterInterface.php (hunk 2 above)
Comment (Simperfit) [reply src/Symfony/Component/Ldap/Adapter/AdapterInterface.php]: fixed :)

**Comment by Simperfit** [comment]
Status: Needs Review

File: src/Symfony/Component/Ldap/Adapter/AdapterInterface.php [hunk 3]
```diff
@@ -32,7 +32,7 @@ public function getConnection();
      *
      * @return QueryInterface
      */
-    public function createQuery($dn, $query, array $options = []);
+    public function createQuery(string $dn, string $query, array $options = []);
```
Comment (derrabus) [inline src/Symfony/Component/Ldap/Adapter/AdapterInterface.php]: We can remove the `@param` annotations here and from a quite a few of other methods as well.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Ldap/Adapter/AdapterInterface.php (hunk 3 above)
Comment (Simperfit) [reply src/Symfony/Component/Ldap/Adapter/AdapterInterface.php]: fixed

**Comment by Simperfit** [comment]
Status: Needs Review

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Ldap/Adapter/AdapterInterface.php [hunk 4]
```diff
@@ -44,11 +40,7 @@ public function getEntryManager();
...
      * Escape a string for use in an LDAP filter or DN.
      *
-     * @param string $subject
-     * @param string $ignore
-     * @param int    $flags
-     *
      * @return string
      */
-    public function escape($subject, $ignore = '', $flags = 0);
+    public function escape(string $subject, ?string $ignore = '', int $flags = 0);
```
Comment (fabpot) [inline src/Symfony/Component/Ldap/Adapter/AdapterInterface.php]: Are we sure we want to support `null` here for `$ignore`? There is a test doing that, but the current phpdoc does not document it. Passing an empty string is enough, right? `ldap_escape` does not seem to support `null` explicitly.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Ldap/Adapter/AdapterInterface.php (hunk 4 above)
Comment (Simperfit) [reply src/Symfony/Component/Ldap/Adapter/AdapterInterface.php]: I agree. This is fixed.

Comment (fabpot) [reply src/Symfony/Component/Ldap/Adapter/AdapterInterface.php]: Should  be changed everywhere.

Comment (Simperfit) [reply src/Symfony/Component/Ldap/Adapter/AdapterInterface.php]: done

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Simperfit.

**Pull request merged**